tests/components/vodafone_station/test_config_flow.py

```diff
@@ -0,0 +1,158 @@
+        assert result["step_id"] == "user"
+        assert result["errors"]["base"] == error
+
+
+async def test_reauth_successful(hass: HomeAssistant) -> None:
+    """Test starting a reauthentication flow."""
+
+    mock_config = MockConfigEntry(domain=DOMAIN, data=MOCK_USER_DATA)
+    mock_config.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.vodafone_station.config_flow.VodafoneStationApi.login",
+    ), patch(
+        "homeassistant.components.vodafone_station.config_flow.VodafoneStationApi.logout",
+    ), patch(
+        "homeassistant.components.vodafone_station.async_setup_entry"
+    ), patch(
+        "requests.get"
+    ) as mock_request_get:
+        mock_request_get.return_value.status_code = 200
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_REAUTH, "entry_id": mock_config.entry_id},
+            data=mock_config.data,
+        )
+
+        assert result["type"] == FlowResultType.FORM
+        assert result["step_id"] == "reauth_confirm"
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={
+                CONF_PASSWORD: "other_fake_password",
+            },
+        )
+        await hass.async_block_till_done()
+
+        assert result["type"] == FlowResultType.ABORT
+        assert result["reason"] == "reauth_successful"
+
+
+@pytest.mark.parametrize(
+    ("side_effect", "error"),
+    [
+        (aiovodafone_exceptions.CannotConnect, "cannot_connect"),
+        (aiovodafone_exceptions.CannotAuthenticate, "invalid_auth"),
+        (ConnectionResetError, "unknown"),
+    ],
+)
+async def test_reauth_not_successful(hass: HomeAssistant, side_effect, error) -> None:
+    """Test starting a reauthentication flow but no connection found."""
+
+    mock_config = MockConfigEntry(domain=DOMAIN, data=MOCK_USER_DATA)
+    mock_config.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.vodafone_station.config_flow.VodafoneStationApi.login",
+        side_effect=side_effect,
+    ), patch(
+        "homeassistant.components.vodafone_station.config_flow.VodafoneStationApi.logout",
+    ), patch(
+        "homeassistant.components.vodafone_station.async_setup_entry"
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_REAUTH, "entry_id": mock_config.entry_id},
+            data=mock_config.data,
+        )
+
+        assert result["type"] == FlowResultType.FORM
+        assert result["step_id"] == "reauth_confirm"
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={
+                CONF_PASSWORD: "other_fake_password",
+            },
+        )
+
+        assert result["type"] == FlowResultType.FORM
+        assert result["step_id"] == "reauth_confirm"
+        assert result["errors"]["base"] == error
```

Comment:
Please finish the flow so it reaches a CREATE_ENTRY or ABORT so we know it can recover from error

Reply:
Thx a lot for the help, done